### PR TITLE
Always use DatasetAddress in `IEXEC_DATASET_FILENAME` env variable

### PR DESCRIPTION
**Pull request by thewhitewizard**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,8 @@
 All notable changes to this project will be documented in this file.
 
 ## [[NEXT]](https://github.com/iExecBlockchainComputing/iexec-common/releases/tag/vNEXT) 2023
-
+### Bug Fixes
+- Use always DatasetAddress in `IEXEC_DATASET_FILENAME` environment variable. (#416)
```
Comment (mcornaton) [inline CHANGELOG.md]:
```suggestion
- Always use DatasetAddress in `IEXEC_DATASET_FILENAME` environment variable. (#416)
```

File: src/main/java/com/iexec/common/utils/IexecEnvUtils.java [hunk 2]
```diff
@@ -97,7 +97,7 @@ public static Map<String, String> getComputeStageEnvMap(TaskDescription taskDesc
         map.put(IEXEC_OUT, IexecFileHelper.SLASH_IEXEC_OUT);
         // dataset
         map.put(IEXEC_DATASET_ADDRESS, taskDescription.getDatasetAddress());
```
Comment (mcornaton) [inline src/main/java/com/iexec/common/utils/IexecEnvUtils.java:99]:
```suggestion
```
This is now unused.

File: src/test/java/com/iexec/common/utils/IexecEnvUtilsTest.java [hunk 3]
```diff
@@ -83,7 +83,7 @@ void getComputeStageEnvList() {
                         "IEXEC_IN=/iexec_in",
                         "IEXEC_OUT=/iexec_out",
                         "IEXEC_DATASET_ADDRESS=datasetAddress",
```
Comment (mcornaton) [inline src/test/java/com/iexec/common/utils/IexecEnvUtilsTest.java:85]:
```suggestion
```
If we remove this from `IexecEnvUtils#getComputeStageEnvMap`, we have to remove it here too.

Comment (jbern0rd) [reply src/test/java/com/iexec/common/utils/IexecEnvUtilsTest.java:85]: After discussion, we will clean this in a dedicated PR

File: src/main/java/com/iexec/common/utils/IexecEnvUtils.java (hunk 2 above)
Comment (jbern0rd) [reply src/main/java/com/iexec/common/utils/IexecEnvUtils.java:99]: After discussion, we will clean this in a dedicated PR

**Review by mcornaton — APPROVED** [review]

**Pull request merged**